op-e2e/system_test.go

```diff
@@ -1098,7 +1098,9 @@ func TestWithdrawals(t *testing.T) {
 	// Fun fact, the fee is greater than the withdrawal amount
 	// NOTE: The gas fees include *both* the ProveWithdrawalTransaction and FinalizeWithdrawalTransaction transactions.
 	diff = new(big.Int).Sub(endBalance, startBalance)
-	fees = calcGasFees(proveReceipt.GasUsed+finalizeReceipt.GasUsed, tx.GasTipCap(), tx.GasFeeCap(), header.BaseFee)
```

Comment:
Here, in the original code, we are invoking `calcGasFees` with `proveReceipt` (L1), `finalizeReceipt` (L1), and the gas parms froom `tx` (L2).

So, when the SuggestedTipPrice for the L2 Tx, and the L1 Tx differ, (if using Erigon for the L2, and Geth for the L1), then `calcGasFees` will return the L1 Gas Usage * the L2 Gas cost, which is wrong.

Reply:
I can't comment there as it's not part of the diff, but line 1055 is where `tx` is set (and it is never overridden), and it's an L2 transaction.

We use it in `calcGasFees` on line 1073 with the receipt from L2 which is correct.  Then, again here at 1101 with the L1 receipts, which is the bug.